src/renderers/dom/shared/dangerousStyleValue.js

```diff
@@ -48,7 +51,27 @@ function dangerousStyleValue(name, value) {
   }
 
   if (typeof value === 'string') {
-    value = value.trim();
+    if (__DEV__) {
+      if (component) {
+        var owner = component._currentElement._owner;
+        var componentName = owner ? owner.getName() : component._currentElement.type;
+        if (!styleWarnings[componentName]) {
+          styleWarnings[componentName] = {};
+        }
+        var warnings = styleWarnings[componentName];
+        if (!warnings[name]) {
+          warnings[name] = true;
+          warning(
+            false,
+            '`%s` passed a numeric string value for CSS property `%s` ' +
+            'which will be treated as a unitless number.',
+            componentName,
+            name
+          );
+        }
+      }
+    }
+    return value.trim();
   }
   return value + 'px';
```

Comment:
We should only append 'px' if the value can be parsed as an integer.  If they specify units, we should honor those units to avoid printing '5pxpx' or '5empx' which would make no sense.  We can make that change in 0.15 (ie. now), since it should break any old code since presumably no one is doing that yet (because the old code would break).